modules/apps/app-builder/app-builder-test/src/testFunctional/macros/FormViewBuilder.macro

```diff
@@ -22,11 +22,24 @@ definition {
 	}
 
 	macro addFieldByDoubleClickFromObjectSidebar {
-		var key_fieldLabel = "${fieldLabel}";
+		var key_fieldLabel = "${fieldType}";
+
+		var iterator = "0";
+		var maxRetryNumber = "3";
+
+		while ((IsElementNotPresent(locator1 = "FormViewBuilder#SIDEBAR_BASIC_TAB")) && ("${iterator}" != "${maxRetryNumber}")) {
+			DoubleClick(locator1 = "FormViewBuilder#OBJECT_FIELD_LABEL");
+
+			Pause(locator1 = "1000");
+
+			if ("${iterator}" == "${maxRetryNumber}") {
+				echo("Couldn't add the field by double-click action, maximum retries number reached: ${iterator}");
+			}
```

Comment:
Please, refactor this part of the code following this example: https://github.com/rodrigocunhaa/liferay-portal/pull/45/files